Geometry/HcalTowerAlgo/src/HcalGeometry.cc

```diff
@@ -163,6 +163,23 @@ GlobalPoint HcalGeometry::getBackPosition(const DetId& id) const {
   }
 }
 
+CaloCellGeometry::CornersVec HcalGeometry::getCorners(const DetId& id) const {
+  if (!m_mergePosition) {
+    return (getGeometry(id)->getCorners());
+  } else {
+    std::vector<HcalDetId> ids;
+    m_topology.unmergeDepthDetId(HcalDetId(id),ids);
+    CaloCellGeometry::CornersVec mcorners;
+    CaloCellGeometry::CornersVec mcf = getGeometry(ids.front())->getCorners();
+    CaloCellGeometry::CornersVec mcb = getGeometry(ids.back())->getCorners();
+    for (unsigned int k=0; k<4; ++k) {
```

Comment:
hello @bsunanda 

can you guarantee (or do you need to check)
that mcf/mcb have at least , resp. , 4 and 8 elements ?